docs/src/HOWTOs/howto-write-type-annotations.md

```diff
@@ -0,0 +1,347 @@
+# How to write type annotations
+
+**Warning:** *This HOWTO discusses how to write type annotations for the new
+type checker [Snowcat][], which has not been integrated with the model checker
+yet.  If you want to run the model checker, you have to write the [old type
+annotations][], until we release the integration (soon!)* :(
+
+This HOWTO gives you concrete steps to extend TLA+ specifications with type
+annotations. You can find the detailed syntax of type annotations in
+[ADR002][]. The first rule of writing type annotations:
+
+*Do not to write any annotations at all, until the type checker [Snowcat][] is
+asking you to write a type annotation.*
+
+Snowcat is quite smart, and it will most probably find most of the types for
+you.
```

Comment:
Given the very first example, where It cannot deduce that `hr` is a nat even tho it is clearly a nat from the very next line, I think this line may give unrealistic expectations.

Reply:
That's probably true. The problem is that declarations are analyzed in isolation. That is good and bad. But sure, I will rephrase it.